design/external-references.md

```diff
@@ -144,18 +154,22 @@ The deserialization context is an object that implements the following interface
 interface IDeserializationContext {
   scope: Construct;
   id: string;
-  readString(key: string): string;
+  readString(key: string, options?: ReadStringOptions): string;
   readObject(key: string): IDeserializationContext;
```

Comment:
We still need lists; I'd rather not leave it to users to have to manually split and join if we can do the same.

Deployment-time lists would work by `{Fn::Join}/{Fn::Split}`, synthesis-time lists could do the same eagerly.

Reply:
We will need to escape the delimiter and I can't see a safe way to do this with deployment-time imports. Maybe we can say that readList() is always `allowUnresolved: false`.

Reply:
We can also say "list values are not allowed to contain this string: `||`" (or some other unlikely separator). I think this is defensible.

* We don't handle arbitrary data, there's a limited set of things that pass around this way (IDs, ARNs, names).
* If construct authors pass around data they're worried will contain these separators, they can always drop down to doing it themselves with `readString()`.
* Not dealing with it at the framework level doesn't make the need go away, it just pushes the concern to construct authors who (a) are all going to write similar code and (b) might write it poorly.